codegen/parser.py

```diff
@@ -272,6 +272,18 @@ class PrimitiveField(_BaseField):
     default: str | int | float | bool | None
 
     def is_nullable(self, version: int) -> bool:
+        # Primitive types are never optional
```

Comment:
thought (non-blocking): This is interesting, I wonder whether this means that we can theoretically treat the combination of one of those types together with `nullableVersions` as a parse error when parsing schema.

Reply:
Probably so. Kafka [does this](https://github.com/apache/kafka/blob/5b295293c06f80accb86df5c46f78c1e5629f86a/generator/src/main/java/org/apache/kafka/message/FieldSpec.java#L95) (`canBeNullable` if `false` for primitives).